librz/core/cmd_system.c

```diff
@@ -143,7 +143,7 @@ static bool system_exec(RzCore *core, int argc, const char **argv, char **output
 		.envvals = envvals,
 		.env_size = RZ_ARRAY_SIZE(envvars),
 		.stdin_pipe = RZ_SUBPROCESS_PIPE_NONE,
-		.stdout_pipe = output ? RZ_SUBPROCESS_PIPE_CREATE : RZ_SUBPROCESS_PIPE_NONE,
+		.stdout_pipe = force_output || core->is_pipe ? RZ_SUBPROCESS_PIPE_CREATE : RZ_SUBPROCESS_PIPE_NONE,
```

Comment:
isn't easier to check for the output ptr?

Reply:
I was not sure it is possible to do that way, because when you execute `!<cmd>` in some cases you want the output, in others you don't want it. So from rz_system_handler you have to always pass the `output` var.

Reply:
yes, but the output needs to go somewhere and return it, so the output pointer MUST be non NULL.

Reply:
i would move the core->is_pipe check one function above, then pass the pointer accordingly to the check.
i mean, this way also works.